### ISSUE TITLE
images on private repository are not found after upgrade to swarm 1.2.1

### ISSUE THREAD

**Issue by scareything** [docker/swarm#2217]
Image pulls through swarm started failing when we introduced 1.2.1 into our environment.

```
$ docker exec d83f21545728 /swarm -v
swarm version 1.2.1 (9a968f5)

$ docker -H 0.0.0.0:4000 pull repo.twx.io/3p-app/dse:8.0.0-latest
ip-10-42-3-6: Pulling repo.twx.io/3p-app/dse:8.0.0-latest... : Error: image 3p-app/dse not found
ip-10-42-3-7: Pulling repo.twx.io/3p-app/dse:8.0.0-latest... : Error: image 3p-app/dse not found
```

Rolling back to 1.2.0 resolves the issue:
```
$ docker exec -it e9ddd1b3c73a /swarm -v
swarm version 1.2.0 (a6c1f14)

$ docker -H 0.0.0.0:4000 pull repo.twx.io/3p-app/dse:8.0.0-latest
ip-10-42-3-6: Pulling repo.twx.io/3p-app/dse:8.0.0-latest... : downloaded 
ip-10-42-3-7: Pulling repo.twx.io/3p-app/dse:8.0.0-latest... : downloaded 
```

**Comment by koosc** [comment]
Not sure if related, but 1.2.1 caused pulling images from our private registry that requires authentication to give a 403 error when using the remote API.

**Comment by Lukasz-Wisniewski** [comment]
We also observe such behaviour after update to swarm 1.2.1 (images from private repository couldn't be downloaded).

Pulling directly through nodes works fine.

Rollbacking to 1.2.0 solved the issue.

**Comment by robvelor** [comment]
Might be duplicate of https://github.com/docker/swarm/issues/1484, rolling back to `1.2.0` seems to solve the issue.

**Comment by bargenson** [comment]
Same issue with the 1.2.2.
I get a `unauthorized: authentication required ` at each pull through the Swarm master but it works perfectly fine if I do it from the nodes directly.

**Comment by jeffjen** [comment]
Confirmed problem against Amazon ECR

Swarm 1.2.2 Having trouble pull image from ECR even though login was performed

Swarm 1.2.0 works.

**Comment by garagatyi** [comment]
Pull on build is also broken because swarm doesn't pass auth config to docker. #2256

**Comment by noose** [comment]
I can pull on `1.2.0` but can't push repos (`unauthorized: authentication required`).
I need to build images on CI. When I build one of it I want to push it and pull on other cluster nodes but I can't.

**Comment by nishanttotla** [comment]
`1.2.3-rc1` that we released yesterday should resolve the issue of pulls that require authentication. This was a bug and the fix was part of #2272, so please let us know if this works now.

**Comment by robvelor** [comment]
@nishanttotla I'll be trying this soon and report back :+1:

**Comment by nishanttotla** [comment]
Thanks @robvelor

**Comment by scareything** [comment]
Thanks for the ping @nishanttotla. I'm running 1.2.3-rc1 now and it's looking good.

**Comment by nishanttotla** [comment]
@scareything great! If no problems are reported by others, I'll consider this issue resolved.

**Comment by robvelor** [comment]
@nishanttotla Looking good, no issues pulling images from a private registry into a swarm with `1.2.3-rc1`.

**Comment by nishanttotla** [comment]
Closing this issue since it was resolved in #2272 and now confirmed. Feel free to ping if the fix isn't working for you.
